hack/verify-typecheck.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+kube::golang::verify_go_version
+
+cd "${KUBE_ROOT}"
+
+time go run test/typecheck/main.go "$@"
```

Comment:
Two comments:
- `time` is redundant here. The verify job times and reports for all verify scripts
- Should this run as a part of the verify job, or should this be a short circuit on the cross job (such as, if this script detects a type check failure, it triggers a full cross build to verify)? I think the latter is what was talked about in Slack, but I'm open to it being in the very job with how quickly it runs. My main concern here is I don't necessarily want to break every PR because of a cross build failure.

Reply:
Removed the time line.

Running it as a verify job works to prevent global breakages, since that's already running for every PR. We were actually talking about running this *before* each build (for just the platform it's running on), spending <30s to catch build errors faster.